src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -203,6 +203,21 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
     }
   }
 
+  /**
+    * Overwrites an existing policy's membership (keyed by resourceType/resourceId/policyName) if it exists
+    * @param resourceType
+    * @param policyName
+    * @param resource
+    * @param membersList
+    * @return
+    */
+  def overwritePolicyMembers(resourceType: ResourceType, policyName: AccessPolicyName, resource: Resource, membersList: Set[WorkbenchEmail]): Future[AccessPolicy] = {
+    loadResourcePolicy(ResourceAndPolicyName(resource, policyName)).flatMap {
+      case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Policy not found"))
+      case Some(accessPolicyMembership) => overwritePolicy(resourceType, policyName, resource, accessPolicyMembership.copy(memberEmails = membersList))
```

Comment:
this is a race condition waiting to happen: thread 1 calls overwritePolicyMembers and loads the policy, thread 2 then overwrites the policy, updating roles (or something other than members), then thread 1 overwrites the policy with the old roles.

It would better if this only updates what it needs to. And no need to load first.